framework/Optimizers/GradientDescent.py

```diff
@@ -598,35 +599,42 @@ def _checkAcceptability(self, traj, opt, optVal, info):
         acceptable = 'accepted'
       else:
         if self._impConstraintFunctions:
-          accept = self._handleImplicitConstraints(old)
+          accept = self._handleImplicitConstraints(opt)
           if accept:
-            acceptable = self._checkForImprovement(optVal, oldVal)
+            acceptable, rejectReason = self._checkForImprovement(optVal, oldVal)
           else:
             acceptable = 'rejected'
+            rejectReason = 2
```

Comment:
as above... the integer is not explicable... I do not know what it means if I dont read the method.